ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -126,16 +126,11 @@ private void waitForCompactionToFinish(CompactionResponse resp) throws HiveExcep
         context.getConsole().printInfo("Interrupted while waiting for compaction with id=" + resp.getId());
         break;
       }
-
-      //this could be expensive when there are a lot of compactions....
-      //todo: update to search by ID once HIVE-13353 is done
-      ShowCompactResponse allCompactions = context.getDb().showCompactions();
-      for (ShowCompactResponseElement compaction : allCompactions.getCompacts()) {
-        if (resp.getId() != compaction.getId()) {
-          continue;
-        }
-
-        switch (compaction.getState()) {
+      ShowCompactRequest request = new ShowCompactRequest();
+      request.setId(resp.getId());
+      ShowCompactResponse compaction = context.getDb().showCompactions(request);
+      if (compaction.getCompactsSize() == 1) {
+        switch (compaction.getCompacts().get(0).getState()) {
```

Comment:
please create a local var for `compaction.getCompacts().get(0)`

Reply:
done